Source/Core/Core/IOS/Network/WD/Command.cpp

```diff
@@ -8,29 +8,418 @@
+      break;
+    case LinkState::Status::_19:
+      m_link_state.status = LinkState::Status::_19;
+      break;
+    case LinkState::Status::_32:
+    case LinkState::Status::_33:
+      m_link_state.status = LinkState::Status::_32;
+      break;
+    case LinkState::Status::_48:
+    case LinkState::Status::_49:
+      m_link_state.status = LinkState::Status::_48;
+      break;
+    case LinkState::Status::_50:
+      m_link_state.status = LinkState::Status::_50;
+      break;
+    case LinkState::Status::Ready:
+    case LinkState::Status::Reset:
+      break;
+    }
+    break;
+
+  case LinkState::Status::Scanning:
+    if (target_status == LinkState::Status::Scanning2)
+      m_link_state.status = LinkState::Status::Scanning2;
+    else
+      m_link_state.status = LinkState::Status::Ready;
+    break;
+
+  case LinkState::Status::Scanning2:
+    m_link_state.status = LinkState::Status::Scanning;
+    break;
+
+  case LinkState::Status::_18:
+  case LinkState::Status::_19:
+    m_link_state.status = LinkState::Status::Ready;
+    break;
+
+  case LinkState::Status::_32:
+    if (target_status == LinkState::Status::_33)
+      m_link_state.status = LinkState::Status::_33;
+    else
+      m_link_state.status = LinkState::Status::Ready;
+    break;
+
+  case LinkState::Status::_33:
+    m_link_state.status = LinkState::Status::_32;
+    break;
+
+  case LinkState::Status::_48:
+    if (target_status == LinkState::Status::_49)
+      m_link_state.status = LinkState::Status::_49;
+    else
+      m_link_state.status = LinkState::Status::Ready;
+    break;
+
+  case LinkState::Status::_49:
+    m_link_state.status = LinkState::Status::_48;
+    break;
+
+  case LinkState::Status::_50:
+  case LinkState::Status::Reset:
+    m_link_state.status = LinkState::Status::Ready;
+    break;
+
+  default:
+    break;
+  }
+
+  INFO_LOG_FMT(IOS_NET, "{}: done (status: {} -> {}, target was {})", __func__, status,
+               m_link_state.status, target_status);
+}
+
+void NetWDCommand::DoState(PointerWrap& p)
+{
+  Device::DoState(p);
+  p.Do(m_ipc_flags);
+  p.Do(m_ipc_fd_map);
+  p.Do(m_mode);
+  p.Do(m_buffer_flags);
+  p.Do(m_link_state);
+  p.Do(m_nitro_enabled_channels);
+  p.Do(m_info);
+  p.Do(m_recv_frame_requests);
+  p.Do(m_recv_notification_requests);
+}
+
+IPCCommandResult NetWDCommand::Open(const OpenRequest& request)
+{
+  if (m_ipc_flags < 0)
+  {
+    const auto flags = u32(request.flags);
+    const auto mode = WD::Mode(flags & 0xFFFF);
+    const auto buffer_flags = flags & 0x7FFF0000;
+    INFO_LOG_FMT(IOS_NET, "Opening with mode={} buffer_flags={:08x}", mode, buffer_flags);
+    bool is_valid_mode = false;
+    if (m_link_state.target_status == LinkState::Status::Ready)
+    {
+      // TODO: check that state.status is not Ready
+      if (mode <= WD::Mode::_6)
+      {
+        m_mode = mode;
+        is_valid_mode = true;
+      }
+    }
+
+    if (is_valid_mode)
+    {
+      m_ipc_flags = flags + 1;
```

Comment:
Any idea why `+1`?

Reply:
Probably a bit/flag used to indicate it's open or ready I guess.

Otherwise, its not uncommon to have state flags to be incremented in a setup/negotiation process when it enters the next step of the process.

Reply:
Yeah. The weird thing is that this performs an addition and not a bitwise OR though...